go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -932,6 +937,105 @@ func (call builtinTrim) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.VarChar, Flag: flagNullable, Col: col}, nil
 }
 
+func (call *builtinSubstring) eval(env *ExpressionEnv) (eval, error) {
+	str, err := call.Arguments[0].eval(env)
+	if err != nil || str == nil {
```

Comment:
Not sure what you mean here? We can't lose the error here?

Reply:
Hypothetically a return from eval of `(nil, nil)` would mean that a nil eval and nil error get returned from the function. Is that okay?

I just mention it because I notice all calls to eval have to check both values, which means that it may be possible to simplify a bit by only having the result be valid when err != nil. That would be a larger refactor.

Reply:
@mdlayher Yep, `nil, nil` is ok. A SQL `NULL` inside the evalengine is represented as the Go `nil`. So it's a perfectly possible and valid value here.